Module: added Confiant RTD adapter

denys-berzoy-confiant opened the pull request:
## Type of change
- [x] Feature
- [x] Does this change affect user-facing APIs or examples documented on http://prebid.org?

## Description of change
Added Confiant’s module that provides comprehensive detection of security, quality, and privacy threats across the ad stack. Confiant is the industry leader in real-time detecting and blocking of bad ads when it comes to protecting your users and brand reputation.

garygz-confiant commented on modules/confiantRtdProvider.js:
```diff
@@ -0,0 +1,80 @@
+    const isConfiantMonitoredEvent = e.data?.type === 'confiant:monitored';
+    if (isConfiantMonitoredEvent) {
+      events.emit(CONSTANTS.EVENTS.BILLABLE_EVENT, {
+        auctionId: e.data.auctionId,
```
once we extend to covering GPT - I am wondering what we would pass for auctionId

garygz-confiant replied:
or transactionId

garygz-confiant commented on modules/confiantRtdProvider.js:
```diff
@@ -0,0 +1,80 @@
+ * Emit billable event when Confiant integration monitors an impression
+ */
+function reportBillableEvents () {
+  window.addEventListener('message', (e) => {
```
gpt confiatDfwWrap can execute from inside the iframe not the same window as confiantWrap - hod we account for that confiatDfwWrap?

garygz-confiant replied:
should we raname our adReporter fram to cnftComm frame and start using it for more use cases?

garygz-confiant commented on modules/confiantRtdProvider.js:
```diff
@@ -0,0 +1,80 @@
+ */
+function reportBillableEvents () {
+  window.addEventListener('message', (e) => {
+    const isConfiantMonitoredEvent = e.data?.type === 'confiant:monitored';
```
we were using this prefix: `cnft:` for events not to be obvious in case hackers attempt to monitor them, see 'cnft:getCmpHealthObj', 'cnft:receiveCmpHealthObj' 'cnft:getAdId' etc.

garygz-confiant commented on modules/confiantRtdProvider.js:
```diff
@@ -0,0 +1,119 @@
+ * @param {Window} targetWindow window instance to subscribe to
+ */
+function subscribeToConfiantCommFrame(targetWindow) {
+  targetWindow.addEventListener('message', reportBillableEvents);
```
add window.confiant.settings.isMGBL = true
to enable billable events

garygz-confiant replied:
if there are multiple confiant stacks on the page we might need to enable for the appropriate config object based on the property id

garygz-confiant commented on modules/confiantRtdProvider.js:
```diff
@@ -0,0 +1,119 @@
+ * @param {Object} config
+ */
+function setupPage(config) {
+  const propertyId = config?.params?.propertyId;
```
we would also want to support one more key:
prebidExcludeBidders
which is a comma separated list of bidders which we should scan
which should be setup as
config.prebidExcludeBidders
e.g. "pubmatic,tryplelift"
This is an optional config key and does not have to be provided.

garygz-confiant replied:
prebidNameSpace is one more key which does not have to implemented now
but will need to be supported in the future
correspond to config.prebidNameSpace
if it's less time let's implement as optional

denys-berzoy-confiant committed:
Confiant's RTD Provider Module

patmmccann committed:
Merge branch 'master' into feature/confiantRtdProvider

denys-berzoy-confiant committed:
Confiant RTD Module:
- updated script injection code to current standard
- added Confiant as an exclusion to load external JS

denys-berzoy-confiant committed:
Merge branch 'prebid-master' into feature/confiantRtdProvider

garygz-confiant commented on modules/confiantRtdProvider.md:
```diff
@@ -34,9 +34,10 @@ pbjs.setConfig({
                 prebidExcludeBidders: '', // optional, comma separated list of bidders to exclude from Confiant's prebid.js integration
-                prebidNameSpace: '', // optional, name space for prebid.js integration
+                prebidNameSpace: '', // optional, string param, namespace for prebid.js integration
+                shouldEmitBillableEvent: false, // optional, boolean param, upon being set to true enables firing BillingEvent upon Confiant's impression scanning
```
in constants  the event is referred to as   "BILLABLE_EVENT": "billableEvent"
not BillingEvent
I think we should keep the language consistent

garygz-confiant reviewed (changes requested):
minor change in comments

garygz-confiant commented:
good to submit to Prebid.js

garygz-confiant commented:
I'll update McCann

denys-berzoy-confiant committed:
Confiant RTD Provider:
- additional param for enabling BillingEvent added
- docs updated
- outdated unit test removed

denys-berzoy-confiant committed:
Confiant RTD provider:
- fix comment
